crates/proof-of-sql/src/sql/proof_plans/group_by_exec.rs

```diff
@@ -198,14 +198,27 @@ impl ProverEvaluate for GroupByExec {
     #[tracing::instrument(name = "GroupByExec::result_evaluate", level = "debug", skip_all)]
     fn result_evaluate<'a, S: Scalar>(
         &self,
-        input_length: usize,
         alloc: &'a Bump,
         accessor: &'a dyn DataAccessor<S>,
     ) -> Vec<Column<'a, S>> {
+        let column_refs = self.get_column_references();
+        let used_table = if column_refs.is_empty() {
+            //TODO: Currently we have to have non-empty column references to have a non-empty table
+            // to evaluate `ProofExpr`s on. Once we restrict [`DataAccessor`] to [`TableExec`]
+            // and use input `DynProofPlan`s we should no longer need this.
```

Comment:
Might be wise to add `row_count` as a field of `Table`. This would fix this issue as well as the comments I have above. See [RecordBatch](https://github.com/apache/arrow-rs/blob/3ee5048c8ea3aa531d111afe33d0a3551eabcd84/arrow-array/src/record_batch.rs#L76-L79).